cpp/src/arrow/util/align_util_test.cc

```diff
@@ -59,6 +63,18 @@ void CheckBitmapWordAlign(const uint8_t* data, int64_t bit_offset, int64_t lengt
   }
 }
 
+arrow::Result<std::shared_ptr<Array>> CreateUnalignedArray(std::shared_ptr<Array> array) {
+  BufferVector sliced_buffers(array->data()->buffers.size(), nullptr);
+  for (std::size_t i = 0; i < array->data()->buffers.size(); ++i) {
+    if (array->data()->buffers[i]) {
+      sliced_buffers[i] = array->data()->buffers[i]->CopySlice(0, 2).MoveValueUnsafe();
```

Comment:
Sorry, this comment was meant to accompany the review but it seems Github swallowed it.  `CopySafe` allocates a new buffer and copies the slice into this new buffer.  The newly allocated buffer will have at least 64 byte alignment but, depending on the system allocator (or just bad luck), this alignment could be greater.

Instead you should use [`SliceBuffer`](https://github.com/apache/arrow/blob/main/cpp/src/arrow/buffer.h#L322).  However, `SliceBuffer` returns a view only (this is inevitable here) so you need to make sure to keep the original array alive.  To do that change this method to accept `const Array&` instead of `std::shared_ptr<Array>`.  Then you should be ok.

Reply:
made the change, thanks for the suggestion.